src/core/sequence_batch_scheduler.cc

```diff
@@ -149,6 +165,101 @@ SequenceBatchScheduler::Create(
   return Status::Success;
 }
 
+Status
+SequenceBatchScheduler::GenerateInitialStateData(
+    const inference::ModelSequenceBatching_InitialState& initial_state,
+    const inference::ModelSequenceBatching_State& state, TritonModel* model)
+{
+  if (initial_state.data_type() != state.data_type()) {
+    return Status(
+        Status::Code::INVALID_ARG,
+        std::string("The data type used for 'initial_state' field of state '") +
+            state.input_name() + "' does not match the state data type.");
+  }
+
+  if (initial_state.name().size() == 0) {
+    return Status(
+        Status::Code::INVALID_ARG,
+        std::string("Field 'name' must be set when using initial_state for "
+                    "state input '") +
+            state.input_name() + "'.");
+  }
+
+  auto initial_state_itr = initial_state_.find(state.input_name());
+  if (initial_state_itr != initial_state_.end()) {
+    return Status(
+        Status::Code::INVALID_ARG, std::string("State input name '") +
+                                       state.input_name() +
+                                       "' specified more than once.");
+  }
+
+  // Check the dimensions to make sure it doesn't have variable-sized dims.
+  for (auto& dim : initial_state.dims()) {
+    if (dim == -1) {
```

Comment:
I think you should also check if the dim is consistent when the correspondence in both `state` and `initial_state` are not -1?